### ISSUE TITLE
Truncate and fade out the bottom of long posts

### ISSUE THREAD

**Issue by BioTurboNick** [mastodon/mastodon#21047]
### Pitch

Mastodon could determine if a post goes beyond a reasonable line count and collapse it, potentially with a faded effect to make clear that more content is available.

This could be an option, if not a default.

### Motivation

Many posts on Mastodon can be long. Some encourage these should go behind a CW to reduce timeline clutter. But that places the onus on the poster to decide what the reader may want.

With this feature, readers would have more control over what their timeline looks like, and posters won't have to worry as much about timeline clutter.

Additionally, reduces the overuse of CWs to instances where it might really be sensitive content, rather than just a convenience.

**Comment by trwnh** [comment]
There's already a "show more" feature

**Comment by BioTurboNick** [comment]
Where? My feed has posts that take up the whole screen, nary a truncation in sight.

**Comment by trwnh** [comment]
wait, it might be a thing only in the advanced web ui then?

**Comment by BioTurboNick** [comment]
I see infosec.exchange has something like what I asked for here... maybe it's the "glitch-soc" add-on the server uses?

Ah yep, seems to be: https://glitch-soc.github.io/docs/features/collapsible-toots/

**Comment by trwnh** [comment]
No, it's definitely a thing in vanilla Mastodon in both the simple and advanced UIs:

![image](https://user-images.githubusercontent.com/10606431/202855412-e35a3332-8288-4b68-bdc4-0f2f68183de1.png)

**Comment by BioTurboNick** [comment]
Either I've never seen it, or the threshold is way too high.

**Comment by ClearlyClaire** [comment]
Closed as not applicable to upstream Mastodon: the server you are on is using a fork of Mastodon with a different way of handling long posts. Upstream Mastodon truncates and shows “Read more” while glitch-soc relies on “collapsible toots”.

**Comment by BioTurboNick** [comment]
1) I'm **not** on a fork. That server was just an example of a similar behavior.
2) I have never seen this Read More mechanism. Is the threshold just too high?

I would appreciate answers that were applicable to what I have asked rather than have this be closed prematurely.

**Comment by BioTurboNick** [comment]
I've finally seen it in the wild. 

What is it based on? Characters? Lines displayed?

**Comment by ClearlyClaire** [comment]
> 1. I'm **not** on a fork. That server was just an example of a similar behavior.

Oh, sorry, I misunderstood. Reopening.

> What is it based on? Characters? Lines displayed?

It is based on total displayed height, the cutoff seems to be 706px in height at the moment.

**Comment by BioTurboNick** [comment]
> > 1. I'm **not** on a fork. That server was just an example of a similar behavior.
> 
> Oh, sorry, I misunderstood. Reopening.
> 
> > What is it based on? Characters? Lines displayed?
> 
> It is based on total displayed height, the cutoff seems to be 706px in height at the moment.

Thanks! I amended my post to reflect wanting this to be configurable.

**Comment by ThisIsMissEm** [comment]
This seems like a dupe of #20191
